### PR TITLE
test: Tree_Select_2_spec - Added toggle JS assertions before updating trigger functions

### PR DESCRIPTION
**Pull request by NandanAnantharamu**
## Description

1. Spec has been flaky because when JS toggle is on a property in edit mode it should remain after entering edit mode from deployed mode. In this PR, we are going to check if JS mode on for each trigger paths and if not, toggle JS mode before adding trigger functions in each test. 

2. We were asserting external URL(google.com) for `Navigate To` functionality and navigating back from external URL using `BrowserNavigation`  to app was creating flakiness. Now we will be stubbing the URL for same window. 

Link - **Spec 36** https://www.notion.so/appsmith/Cypress-tests-analysis-4090efa5e3064a2e87f262d3c399a339 
            Phase 1

## Summary by CodeRabbit

- **Tests**
	- Enhanced test scenarios to assert the state of JavaScript toggles in the Tree Select widget.
- **New Features**
	- Introduced a method to toggle and verify JavaScript mode in the Property Pane.

### CODE REVIEW

**Review by ApekshaBhosale — DISMISSED** [review]

File: app/client/cypress/e2e/Regression/ClientSide/Widgets/TreeSelect/Tree_Select_2_spec.ts [hunk 1]
```diff
@@ -243,21 +244,38 @@ describe(
...
+      });
+      deployMode.NavigateBacktoEditor();
+      EditorNavigation.SelectEntityByName("Form1", EntityType.Widget);
+      EditorNavigation.SelectEntityByName(
+        "TreeSelect1",
+        EntityType.Widget,
+        {},
+        ["Form1"],
+      );
+      propPane.AssertJSToggleState("onOptionChange", "enabled");
```
Comment (trishaanand) [inline app/client/cypress/e2e/Regression/ClientSide/Widgets/TreeSelect/Tree_Select_2_spec.ts:275]: Why are we asserting on navigating back to editor that JS state is enabled for on Options change? Is that a product requirement?

Comment (trishaanand) [reply app/client/cypress/e2e/Regression/ClientSide/Widgets/TreeSelect/Tree_Select_2_spec.ts:275]: Also in the next test case, we are directly updating on option change field without toggling on the JS mode

File: app/client/cypress/support/Pages/PropertyPane.ts [hunk 2]
```diff
@@ -673,4 +673,22 @@ export class PropertyPane {
...
   }
+
+  public ToggleJS(property: string) {
+    const checkToggleJSIsActive = (prop: any) => {
+      if (prop.hasClass("is-active")) {
+        cy.log("Toggle is active");
+      } else {
+        this.ToggleJSMode(property, true); // Activate the toggle
+        // Rerun the assertion to ensure it's now active
+        cy.get(
```
[the remark this answers is not in the document]
Comment (ApekshaBhosale) [reply app/client/cypress/support/Pages/PropertyPane.ts:684]: yes if we never able to toggle JS mode on successfully @trishaanand

File: app/client/cypress/e2e/Regression/ClientSide/Widgets/TreeSelect/Tree_Select_2_spec.ts [hunk 3]
```diff
@@ -243,21 +244,38 @@ describe(
         ["Form1"],
       );
       propPane.ToggleJSMode("onOptionChange", true);
+      propPane.ToggleJS("onOptionChange");
```
[the remark this answers is not in the document]
Comment (ApekshaBhosale) [reply app/client/cypress/e2e/Regression/ClientSide/Widgets/TreeSelect/Tree_Select_2_spec.ts:247]: i am thinking we should not use the new ToggleJS  function, it is just a wrapper around ToggleJSMode and only for "true" use cases. ToggleJSMode also checks state before enabling/disabling it. Trying to figure out how can we make sure the toggle has happened

**Comment by trishaanand** [comment]
@ApekshaBhosale : The PR title and description needs to be updated. The code changes LGTM.

**Pull request merged**